hollow/src/main/java/com/netflix/hollow/api/client/HollowClientUpdater.java

```diff
@@ -130,6 +176,7 @@ public synchronized boolean updateTo(long version) throws Throwable {
                 metricsCollector.collect(metrics);
             for(HollowConsumer.RefreshListener refreshListener : refreshListeners)
                 refreshListener.refreshFailed(beforeVersion, getCurrentVersionId(), version, th);
+            initialLoad.completeExceptionally(th);
```

Comment:
This might be bad behavior. Originally I left the future unresolved. For consumers whose producers write often, they will have a new update to try that might succeed. Completing exceptionally means that calling code won't ever see that the consumer eventually had a successful initial load.

If a producer writes infrequently, calling code that blocks/has-callback could be blocked/waiting for a very long time or the life of the application.

I'm leaning now on removing this in favor of recovery where producers write often.

Reply:
I agree. It would be nice if we could implement this in an extensible manner with a special refresh listener, then it might be possible to configure behavior, but i suspect order of refresh listener execution may be required to ensure the special refresh listener is invoked after all other refresh listeners.